pkg/core/kafka.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/pkg/errors"
+)
+
+type KafkaAttackAction string
+
+const (
+	// Kafka actions
+	KafkaFillAction  = "fill"
+	KafkaFloodAction = "flood"
+	KafkaIOAction    = "io"
+)
+
+var _ AttackConfig = &KafkaCommand{}
+
+type KafkaCommand struct {
+	CommonAttackConfig
+
+	// global options
+	Action    KafkaAttackAction
+	Topic     string
+	Partition int
+	NoSilent  bool
+
+	// options for fill and flood attack
+	Host        string
+	Port        uint16
+	Username    string
+	Password    string
+	MessageSize uint
+	MaxBytes    uint64
+
+	// options for flood attack
+	Threads          uint
+	RequestPerSecond uint64
+
+	// options for io attack
+	ConfigFile  string
+	NonReadable bool
+	NonWritable bool
+
+	// recover data for io attack
+	PartitionDir string
+}
+
+func (c *KafkaCommand) Validate() error {
```

Comment:
Can we split this function into multiple functions?

Reply:
Split.